src/main/scala/is/hail/methods/LinearMixedRegression.scala

```diff
@@ -151,10 +151,11 @@ object LinearMixedRegression {
 
       vds2.mapAnnotations { case (v, va, gs) =>
         val SparseGtVectorAndStats(x0, isConstant, af, nHomRef, nHet, nHomVar, nMissing) =
-          gs.iterator.zipWithIndex
-            .filter { case (g, i) => sampleMaskBc.value(i)}
-            .foldLeft(new SparseGtBuilder()){ case (b, (g, i)) => b.merge(g) }
-            .toSparseGtVector(n)
+          gs.hardCallIterator
+            .unsafeFilter(sampleMaskBc.value.toIterator)
+            .toBoxedIntIterator
+            .foldLeft(new SparseGtBuilder()){ case (b, gt) => b.merge(gt) }
```

Comment:
The goal is to not do per-genotype allocation.  This allocates multiple times per genotype.